src/bindings.rs

```diff
@@ -125,6 +125,18 @@ pub(crate) mod ffi {
     }
 }
 
+impl ffi::Size {
+    pub fn new(w: u32, h: u32) -> Self {
+        Self { w, h }
+    }
+    pub fn from_dimensions_range(range: &ffi::DimensionsRange) -> Self {
```

Comment:
Sorry to bother you with this but this function may fail if end < start.
You can impl TryFrom<DimensionRange> for Size

Reply:
updated, If you have suggestions on how to better handle the Err do not hesitate 🙂